File: Ransom_Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        counts = [0] * 26
        for c in magazine:
            counts[ord(c) - ord('a')] += 1

        for c in ransomNote:
            idx = ord(c) - ord('a')
            if counts[idx] == 0:
                return False
            counts[idx] -= 1
        return True

File: Ransom_Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] counts = new int[26];
        for (char c : magazine.toCharArray()) {
            counts[c - 'a']++;
        }

        for (char c : ransomNote.toCharArray()) {
            int idx = c - 'a';
            if (counts[idx] == 0) {
                return false;
            } 
            counts[idx]--;
            
        }
        return true;
    }
}